pkg/validation/internal/removed_apis.go

```diff
@@ -2,14 +2,149 @@ package internal
+const k8sVersionKey = "k8s-version"
+
+const minKubeVersionWarnMessage = "csv.Spec.minKubeVersion is not informed. It is recommended you provide this information. " +
+	"Otherwise, it would mean that your operator project can be distributed and installed in any cluster version " +
+	"available, which is not necessarily the case for all projects."
+
+// AlphaDeprecatedAPIsValidator validates if the bundles is using versions API version which are deprecate or
+// removed in specific Kubernetes versions informed via optional key value `k8s-version`.
+var AlphaDeprecatedAPIsValidator interfaces.Validator = interfaces.ValidatorFunc(validateDeprecatedAPIsValidator)
+
+func validateDeprecatedAPIsValidator(objs ...interface{}) (results []errors.ManifestResult) {
+
+	// Obtain the k8s version if informed via the objects an optional
+	k8sVersion := ""
+	for _, obj := range objs {
+		switch obj.(type) {
+		case map[string]string:
+			k8sVersion = obj.(map[string]string)[k8sVersionKey]
+			if len(k8sVersion) > 0 {
+				break
+			}
+		}
+	}
+
+	for _, obj := range objs {
+		switch v := obj.(type) {
+		case *manifests.Bundle:
+			results = append(results, validateDeprecatedAPIs(v, k8sVersion))
+		}
+	}
+
+	return results
+}
+
+func validateDeprecatedAPIs(bundle *manifests.Bundle, k8sVersion string) errors.ManifestResult {
+	result := errors.ManifestResult{Name: bundle.Name}
+
+	if bundle == nil {
+		result.Add(errors.ErrInvalidBundle("Bundle is nil", nil))
+		return result
+	}
+
+	if bundle.CSV == nil {
+		result.Add(errors.ErrInvalidBundle("Bundle csv is nil", bundle.Name))
+		return result
+	}
+
+	errs, warns := validateDeprecatedAPIS(bundle, k8sVersion)
+	for _, err := range errs {
+		result.Add(errors.ErrFailedValidation(err.Error(), bundle.CSV.GetName()))
+	}
+	for _, warn := range warns {
+		result.Add(errors.WarnFailedValidation(warn.Error(), bundle.CSV.GetName()))
+	}
+
+	return result
+}
+
+// validateDeprecatedAPIS will check if the operator bundle is using a deprecated or no longer supported k8s api
+// Note if the k8s was informed via "k8s=1.22" it will be used. Otherwise, we will use the minKubeVersion in
+// the CSV to do the checks. So, the criteria is >=minKubeVersion. By last, if the minKubeVersion is not provided
+// then, we should consider the operator bundle is intend to work well in any Kubernetes version.
+// Then, it means that:
+//--optional-values="k8s-version=value" flag with a value => 1.16 <= 1.22 the validator will return result as warning.
+//--optional-values="k8s-version=value" flag with a value => 1.22 the validator will return result as error.
+//minKubeVersion >= 1.22 return the error result.
+//minKubeVersion empty returns a warning since it would mean the same of allow install in any supported version
+func validateDeprecatedAPIS(bundle *manifests.Bundle, versionProvided string) (errs, warns []error) {
+	// K8s version where the apis v1betav1 is no longer supported
+	const k8sVerV1betav1Unsupported = "1.22.0"
```

Comment:
Can we elevate these two const (`k8sVerV1betav1Unsupported` and `k8sVerV1betav1Deprecated`) to be outside of this function so they can be like `ocpVerV1beta1Unsupported`? And on that note, why do we have an OCP const on an upstream repo? Should we remove `ocpVerV1beta1Unsupported`?

Reply:
This is not a big deal on the k8s const comment but the OCP const is certainly out of place in this repo.